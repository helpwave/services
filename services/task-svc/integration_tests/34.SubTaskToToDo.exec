#!/bin/bash

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "b2ed8f2b-1fc9-4bbe-8705-acf84850c6aa"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/GetTask

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "1e785484-e2a6-4ed5-9212-ee56a9d2c39d"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/SubTaskToToDo

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "b2ed8f2b-1fc9-4bbe-8705-acf84850c6aa"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/GetTask

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "1e785484-e2a6-4ed5-9212-ee56a9d2c39d"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/SubTaskToDone

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "b2ed8f2b-1fc9-4bbe-8705-acf84850c6aa"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/GetTask
