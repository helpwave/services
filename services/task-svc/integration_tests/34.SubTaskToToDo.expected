{
  "id": "b2ed8f2b-1fc9-4bbe-8705-acf84850c6aa",
  "name": "Drainage of Sigmoid Colon with Drain Dev, Perc Endo Approach",
  "description": "Supplement Right Common Iliac Vein with Nonautologous Tissue Substitute, Percutaneous Endoscopic Approach",
  "status": "TASK_STATUS_TODO",
  "assignedUserId": "22b47b87-aadf-4e81-802d-9e23489e696c",
  "patient": {
    "id": "4a2e5163-57a7-4a50-9ed7-981c119848d7",
    "name": "Desiree Ert"
  },
  "dueAt": "2024-11-18T07:54:24Z",
  "createdBy": "8aa8a5ee-2a35-4203-81bb-baf4ca17a9e9",
  "subtasks": [
    {
      "id": "9ea8ff25-6118-4424-99a2-a63a17008f5b",
      "name": "Suture of laceration of anus",
      "createdBy": "73c0d265-3c9a-410a-8eaf-0174fc57f421"
    },
    {
      "id": "1e785484-e2a6-4ed5-9212-ee56a9d2c39d",
      "name": "Laparoscopic oophorotomy",
      "done": true,
      "createdBy": "3e795d31-1912-4b09-8e5a-fd62a2461a8c"
    }
  ],
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406"
}
{}
{
  "id": "b2ed8f2b-1fc9-4bbe-8705-acf84850c6aa",
  "name": "Drainage of Sigmoid Colon with Drain Dev, Perc Endo Approach",
  "description": "Supplement Right Common Iliac Vein with Nonautologous Tissue Substitute, Percutaneous Endoscopic Approach",
  "status": "TASK_STATUS_TODO",
  "assignedUserId": "22b47b87-aadf-4e81-802d-9e23489e696c",
  "patient": {
    "id": "4a2e5163-57a7-4a50-9ed7-981c119848d7",
    "name": "Desiree Ert"
  },
  "dueAt": "2024-11-18T07:54:24Z",
  "createdBy": "8aa8a5ee-2a35-4203-81bb-baf4ca17a9e9",
  "subtasks": [
    {
      "id": "9ea8ff25-6118-4424-99a2-a63a17008f5b",
      "name": "Suture of laceration of anus",
      "createdBy": "73c0d265-3c9a-410a-8eaf-0174fc57f421"
    },
    {
      "id": "1e785484-e2a6-4ed5-9212-ee56a9d2c39d",
      "name": "Laparoscopic oophorotomy",
      "createdBy": "3e795d31-1912-4b09-8e5a-fd62a2461a8c"
    }
  ],
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406"
}
{}
{
  "id": "b2ed8f2b-1fc9-4bbe-8705-acf84850c6aa",
  "name": "Drainage of Sigmoid Colon with Drain Dev, Perc Endo Approach",
  "description": "Supplement Right Common Iliac Vein with Nonautologous Tissue Substitute, Percutaneous Endoscopic Approach",
  "status": "TASK_STATUS_TODO",
  "assignedUserId": "22b47b87-aadf-4e81-802d-9e23489e696c",
  "patient": {
    "id": "4a2e5163-57a7-4a50-9ed7-981c119848d7",
    "name": "Desiree Ert"
  },
  "dueAt": "2024-11-18T07:54:24Z",
  "createdBy": "8aa8a5ee-2a35-4203-81bb-baf4ca17a9e9",
  "subtasks": [
    {
      "id": "9ea8ff25-6118-4424-99a2-a63a17008f5b",
      "name": "Suture of laceration of anus",
      "createdBy": "73c0d265-3c9a-410a-8eaf-0174fc57f421"
    },
    {
      "id": "1e785484-e2a6-4ed5-9212-ee56a9d2c39d",
      "name": "Laparoscopic oophorotomy",
      "done": true,
      "createdBy": "3e795d31-1912-4b09-8e5a-fd62a2461a8c"
    }
  ],
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406"
}
does not exist
{}
{}
